pkg/parser/parser.peg

```diff
@@ -1543,18 +1561,47 @@ ResolvedExternalLink <- url:(ResolvedLocation) inlineAttributes:(LinkAttributes)
 // ------------------------------------------
 // Images
 // ------------------------------------------
-ImageBlock <- attributes:(Attributes)? "image::" path:(Location) inlineAttributes:(ImageAttributes) Space* EOL {
-    return types.NewImageBlock(path.(types.Location), inlineAttributes.(types.Attributes), attributes)
+ImageBlock <- attributes:(BlockImageAttrs)* "image::" path:(Location) inlineAttrs:(InlineImageAttrs) Space* EOL {
+    return types.NewImageBlock(path.(types.Location), inlineAttrs.(types.Attributes), attributes)
+}
+
+InlineImage <- "image:" !":" path:(Location) inlineAttrs:(InlineImageAttrs) {
+    return types.NewInlineImage(path.(types.Location), inlineAttrs.(types.Attributes))
+}
+
+InlineImageAttrs <- '[' attrs:(ImageAltAttrInline? ImageWidthAttr? ImageHeightAttr? NamedAttr*) ']' {
+    return types.NewElementAttributes(attrs)
+}
+
+BlockImageAttrs <- ImageAttrList / ElementTitle / ElementID
+
+ImageAttrList <- '[' attrs:(ImageAltAttr? ShortHandAttr* ImageWidthAttr? ImageHeightAttr? NamedAttr*) ']' Space* EOL {
+    return types.NewElementAttributes(attrs)
 }
 
-InlineImage <- "image:" !":" path:(Location) inlineAttributes:(ImageAttributes) {
-    return types.NewInlineImage(path.(types.Location), inlineAttributes.(types.Attributes))
+ImageAltAttrInline <- value:InlinePositionalValue {
+    return types.NewElementNamedAttr(types.AttrImageAlt, value.(string))
 }
 
-ImageAttributes <- "[" alt:(StandaloneAttributeValue)? ","? width:(StandaloneAttributeValue)? ","? height:(StandaloneAttributeValue)? ","? Space* otherattrs:(GenericAttribute)* "]" {
-    return types.NewImageAttributes(alt, width, height, otherattrs.([]interface{}))
+ImageAltAttr <- Space* value:PositionalValue Space* {
```

Comment:
I assume that the support for spaces around the positional value in the `ImageAltAttr` rule vs lack of support for spaces in the `ImageAltAttrInline` rule is related to the comment below found in the PR?

> There is some strangeness around the acceptance or not of whitespace in attribute lists.

Reply:
You would assume correctly.  I am trying hard not to describe all the ways in which asciidoctor lacks consistency.

Reply:
yes, that one more area in which I hope the WG will clarify!